src/util.rs

```diff
@@ -54,8 +54,9 @@ macro_rules! parse_tag {
 
 /// Creates a new type that wraps an internal data type over along with a map.
 macro_rules! map_wrapper {
-    ($name:ident => $data_ty:ty) => {
+    ($(#[$attrs:meta])* $name:ident => $data_ty:ty) => {
         #[derive(Clone, PartialEq, Debug)]
+        $(#[$attrs])*
```

Comment:
That is some magic that I won't try to understand right now... :-)

Reply:
It's not very complex as long as you know about token repetition with `*`; First we match against any attributes, which are meta elements encased by `#[ ... ]`, and then we repackage them herebin order to serve them as attributes for the wrapper.